rust/lance-io/src/scheduler.rs

```diff
@@ -258,8 +270,61 @@ impl FileScheduler {
     ) -> impl Future<Output = Result<Vec<Bytes>>> + Send {
         // The final priority is a combination of the row offset and the file number
         let priority = ((self.file_index as u128) << 64) + priority as u128;
-        self.root
-            .submit_request(self.reader.clone(), request, priority)
+
+        let mut updated_requests = Vec::with_capacity(request.len());
+
+        let copy_request = request.clone();
+
+        if !request.is_empty() {
+            let mut curr_interval = request[0].clone();
+
+            for req in request.iter().skip(1) {
+                if is_close_together(&curr_interval, req, self.block_size) {
+                    curr_interval.end = curr_interval.end.max(req.end);
+                } else {
+                    updated_requests.push(curr_interval);
+                    curr_interval = req.clone();
+                }
+            }
+
+            updated_requests.push(curr_interval);
+        }
+
+        let copy_updated_requests = updated_requests.clone();
+
+        let bytes_vec_fut =
+            self.root
+                .submit_request(self.reader.clone(), updated_requests, priority);
```

Comment:
nit: I feel like it's a bit clearer if you give the clone rather than the original to the function call.

```suggestion
        let bytes_vec_fut =
            self.root
                .submit_request(self.reader.clone(), updated_requests.clone(), priority);
```